sipn_reanalysis_ingest/luigitasks/convert.py

```diff
@@ -38,67 +44,112 @@ class Grib2ToNcDaily(luigi.Task):
         return luigi.LocalTarget(DATA_FINISHED_DIR / fn)
 
     def run(self):
-        analysis_dir = Path(self.input()[CfsrGranuleProductType.ANALYSIS].path)
-        analysis_inputs = select_6hourly_analysis_grib2s(analysis_dir, date=self.date)
+        if self.uses_5day_tars:
+           analysis_dir = Path(self.input()[CfsrGranuleProductType.ANALYSIS].path)
+           analysis_inputs = select_v1_6hourly_analysis_grib2s(analysis_dir, date=self.date)
 
-        forecast_dirs = [
+           forecast_dirs = [
             Path(d.path)
             for d in luigi.task.flatten(self.input()[CfsrGranuleProductType.FORECAST])
-        ]
-        forecast_inputs = select_6hourly_forecast_grib2s(forecast_dirs, date=self.date)
-
-        logger.info(f'Producing daily NetCDF for date {self.date}...')
-        logger.debug(f'>> Analysis inputs: {analysis_inputs}')
-        logger.debug(f'>> Forecast inputs: {forecast_inputs}')
-
-        with self.output().temporary_path() as tempf:
-            convert_6hourly_grib2s_to_nc(
-                analysis_inputs=analysis_inputs,
-                forecast_inputs=forecast_inputs,
-                output_path=Path(tempf),
-            )
-
-            for ifile in [*analysis_inputs, *forecast_inputs]:
-                ifile.unlink()
-
+           ]
+           forecast_inputs = select_v1_6hourly_forecast_grib2s(forecast_dirs, date=self.date)
+
+           logger.info(f'Producing daily NetCDF for date {self.date}...')
+           logger.debug(f'>> Analysis inputs: {analysis_inputs}')
+           logger.debug(f'>> Forecast inputs: {forecast_inputs}')
+
+           with self.output().temporary_path() as tempf:
+               convert_6hourly_grib2s_to_nc(
+                   analysis_inputs=analysis_inputs,
+                   forecast_inputs=forecast_inputs,
+                   output_path=Path(tempf),
+               )
+
+               for ifile in [*analysis_inputs, *forecast_inputs]:
+                   ifile.unlink()
+        else:
+           analysis_dir = Path(self.input()[CfsrGranuleProductType.ANALYSIS].path)
+           analysis_inputs = select_v2_6hourly_analysis_grib2s(analysis_dir, date=self.date)
+
+           forecast_dir=Path(self.input()[CfsrGranuleProductType.FORECAST].path)
+           forecast_inputs = select_v2_6hourly_forecast_grib2s(forecast_dir, date=self.date)
+
+           logger.info(f'Producing daily NetCDF for date {self.date}...')
+           logger.debug(f'>> Analysis inputs: {analysis_inputs}')
+           logger.debug(f'>> Forecast inputs: {forecast_inputs}')
+
+           with self.output().temporary_path() as tempf:
+               convert_6hourly_grib2s_to_nc(
+                   analysis_inputs=analysis_inputs,
+                   forecast_inputs=forecast_inputs,
+                   output_path=Path(tempf),
+               )
+
+               for ifile in [*analysis_inputs, *forecast_inputs]:
+                   ifile.unlink()
```

Comment:
Can de-duplicate this cleanup step by moving out of the if/else block!

Reply:
Done!